packages/driver/cypress/e2e/commands/cookies.cy.js

```diff
@@ -9,148 +9,142 @@ describe('src/cy/commands/cookies - no stub', () => {
 
-        cy.getCookies({ domain: null }).should('have.length', 2)
+        cy.getCookies({ domain: 'www.barbaz.com' }).then((cookies) => {
+          expect(cookies).to.have.length(3)
+          expect(cookies[0].domain).to.match(/\.?www\.barbaz\.com/)
+          expect(cookies[1].domain).to.match(/\.?www\.barbaz\.com/)
+          expect(cookies[2].domain).to.match(/\.?www\.barbaz\.com/)
+        })
       })
     })
   })
 
   context('#getCookie', () => {
-    it('returns the cookie from the domain matching the AUT by default', () => {
+    beforeEach(() => {
       cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-      cy.setCookie('foo', 'bar')
+      cy.setCookie('key', 'value1', { domain: 'www.foobar.com' })
+      cy.setCookie('key', 'value2')
+    })
 
-      cy.getCookie('foo').its('domain').should('match', /\.?barbaz\.com/)
+    it('returns the cookie from the domain matching the AUT by default', () => {
+      cy.getCookie('key').then((cookie) => {
+        expect(cookie.value).to.equal('value2')
+        expect(cookie.domain).to.match(/\.?barbaz\.com/)
+      })
 
-      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
+      // webkit does not support cy.origin()
+      if (isWebkit) return
 
       cy.origin('http://www.foobar.com:3500', () => {
         cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
 
-        cy.getCookie('foo').its('domain').should('match', /\.?www\.foobar\.com/)
+        cy.getCookie('key').then((cookie) => {
+          expect(cookie.value).to.equal('value1')
+          expect(cookie.domain).to.match(/\.?www\.foobar\.com/)
+        })
       })
     })
 
     it('returns the cookie from the specified domain', () => {
-      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-      cy.setCookie('foo', 'bar')
-
-      cy.getCookie('foo', { domain: 'www.foobar.com' })
-      .its('domain').should('match', /\.?www\.foobar\.com/)
-
-      if (isWebkit || !Cypress.config('experimentalSessionAndOrigin')) return
-
-      cy.origin('http://www.foobar.com:3500', () => {
-        cy.visit('http://www.foobar.com:3500/fixtures/generic.html')
-
-        cy.getCookie('foo', { domain: 'barbaz.com' })
-        .its('domain').should('match', /\.?barbaz\.com/)
+      cy.getCookie('key', { domain: 'www.foobar.com' }).then((cookie) => {
+        expect(cookie.value).to.equal('value1')
+        expect(cookie.domain).to.match(/\.?www\.foobar\.com/)
       })
-    })
 
-    it('returns cookie for any domain when domain is null', () => {
-      cy.visit('http://www.barbaz.com:3500/fixtures/generic.html')
-      cy.setCookie('foo', 'bar', { domain: 'www.foobar.com' })
-
-      cy.getCookie('foo', { domain: null })
-      .its('domain').should('match', /\.?www\.foobar\.com/)
+      // domain must be exact
```

Comment:
should this be it's own it block?
```suggestion
it('returns null when domain does not match', () => {
```